src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -3385,6 +3385,13 @@ GetAggregateType(Aggref *aggregateExpression)
 {
 	Oid aggFunctionId = aggregateExpression->aggfnoid;
 
+	/* custom aggregates with combine func take precedence over name-based logic */
+	if (aggFunctionId >= FirstNormalObjectId &&
```

Comment:
I think this is definitely a good idea & fix.

But, what if `AggregateEnabledCustom` returns false? Wouldn't it be nicer to throw an error in that case? Otherwise, on lucky occasions like pgvector, we'd get operator/function errors. 

But, for some other unlucky ones, it seems we could potentially return wrong results by using wrong combine function?

```C
if (aggFunctionId >= FirstNormalObjectId)
{
   if AggregateEnabledCustom..
   else
      ereport
}
```

Reply:
Might be a bit risky in case users defined their own `sum` aggregates and expected this to work.